caps/management/commands/add_related_councils.py

```diff
@@ -1,41 +1,49 @@
 # -*- coding: future_fstrings -*-
+import os
+import tempfile
+from pathlib import Path
+from shutil import copytree
+
+from caps.models import (ComparisonLabel, ComparisonLabelAssignment,
+                         ComparisonType, Council, Distance)
 from django.core.management.base import BaseCommand, CommandError
+from django.db.models import Count
 
-from caps.mapit import MapIt, NotFoundException, BadRequestException, InternalServerErrorException, ForbiddenException
-from caps.models import Council
+DISTANCE_REPO = "https://github.com/mysociety/la_distance"
 
-from django.db.models import Count
 
-import time
+def download_data():
+    """
+    fetch the current version of the comparisons dataset
+    """
+    dst = Path("data", "comparisons")
+    dst.mkdir(exist_ok=True)
+
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        os.system(f"git clone {DISTANCE_REPO} {tmpdirname}")
+        src = Path(tmpdirname, "data", "outputs")
+        copytree(src, dst, dirs_exist_ok=True)
+
 
 def add_related_councils():
-    mapit = MapIt()
-    # MapIt doesn't currently have information on Combined Authorities
-    councils = Council.objects.exclude(authority_type='COMB')
-    for council in councils:
-        try:
-            mapit_id = mapit.gss_code_to_mapit_id(council.gss_code)
-            time.sleep(2)
-            touching_gss_codes = mapit.mapit_id_to_touches(mapit_id)
-            time.sleep(2)
-            for touching_council in Council.objects.filter(gss_code__in=touching_gss_codes):
-                council.related_councils.add(touching_council)
-            if council.combined_authority:
-                council.related_councils.add(council.combined_authority)
-            council.save()
-        except (NotFoundException, BadRequestException, InternalServerErrorException, ForbiddenException) as error:
-            print(f"Error: {council.name} - '{error}'")
+    ComparisonType.populate()
+    ComparisonLabel.populate()
+    ComparisonLabelAssignment.populate()
+    Distance.populate()
+
 
 class Command(BaseCommand):
-    help = 'Adds geographically related authorities for each council using MapIt'
+    help = 'Adds related authorities for each council using MapIt'
 
     def handle(self, *args, **options):
         get_all = options['all']
-        if get_all:
-            print('adding related authorities')
-            add_related_councils()
-        else:
-            print('skipping related authorities')
+
+        have_data = Path("data", "comparisons").exists()
+        if get_all or have_data is False:
+            print("downloading data")
+            download_data()
+        print('adding related authorities')
+        add_related_councils()
```

Comment:
This will never update the data after the first time (without `get_all`) but will completely re-import everything into the database each time. I think `add_related_councils` should also be inside the if otherwise we're doing a lot of unnecessary database things on each update.

Reply:
Yep, indented so everything is within the all (although this now does nothing without the 'all' , but maybe it will in future?)